src/_igraph/graphobject.c

```diff
@@ -13895,6 +13895,146 @@ PyObject *igraphmodule_Graph_random_walk(igraphmodule_GraphObject * self,
   }
 }
 
+
+/**********************************************************************
+ * Other methods                                                      *
+ **********************************************************************/
+
+/**
+ * Voronoi clustering
+ */
+PyObject *igraphmodule_Graph_community_voronoi(igraphmodule_GraphObject *self,
+                                               PyObject *args, PyObject *kwds) {
+    static char *kwlist[] = {"lengths", "weights", "mode", "radius", NULL};
+    PyObject *lengths_o = Py_None, *weights_o = Py_None;
+    PyObject *mode_o = Py_None;
+    PyObject *radius_o = Py_None;
+    igraph_vector_t *lengths_v = 0;
+    igraph_vector_t *weights_v = 0;
+    igraph_vector_int_t membership_v, generators_v;
+    igraph_neimode_t mode = IGRAPH_ALL;
```

Comment:
I'm leaning towards making `IGRAPH_OUT` the default. This is explicitly designed to handle directed graphs properly. Opinions @ntamas ?

If this is changed, the documentation should also be changed from `"all"` to `"out"`.

Reply:
There isn't really any consistency in the Python interface w.r.t. the defaults of `mode` arguments. Let's see what `functions.yaml` says in the upstream repo and use that default - this will be used in the new Python interface as well.